WcaOnRails/app/models/registration.rb

```diff
@@ -1,10 +1,13 @@
 # frozen_string_literal: true
 class Registration < ActiveRecord::Base
-  scope :pending, -> { where(accepted_at: nil) }
-  scope :accepted, -> { where.not(accepted_at: nil) }
+  scope :pending, -> { where(accepted_at: nil).where(deleted_at: nil) }
+  scope :accepted, -> { where.not(accepted_at: nil).where(deleted_at: nil) }
+  scope :deleted, -> { where.not(deleted_at: nil) }
```

Comment:
How did oyu check this? I just remembered that we do not allow people to edit their profile information if they are registered for a competition, and it looks like this code would be affected by your changes here. See https://github.com/thewca/worldcubeassociation.org/blob/756402ded19b398f3c6cee2dafa915db1e8a1483/WcaOnRails/app/models/user.rb#L476-L478.